File: Subarray_with_given_sum.cpp
#include <iostream>
using namespace std;

void sub_sum(int arr[], int n, int key);

void sub_sum(int arr[], int n, int key)
{
    bool found = false;
    for(int s=0; s<n; s++)
    {
        int sum = 0;
        found = false;
        for(int e=s; e<n; e++)
        {
            sum += arr[e];
            if(sum==key)
            {
                found = true;
                cout<<s+1<<" "<<e+1<<endl;

            }
            if (found == true)
            {
                return;
            }
        }
    }
    if (found == false)
    {
        cout<<"-1"<<endl;
        return;
    }

}

int main()
{
    int t, n, s;

    cin>>t;
    for(int i=0; i<t; i++)
        {
            cin>>n>>s;
            int arr[n];

            for(int j=0; j<n; j++)
            {
                cin>>arr[j];
            }

            sub_sum(arr, n, s);
        }

    return 0;
}

File: subarray_with_given_sum.py
def sub_sum(arr, total):
    n = len(arr)
    result = 0
    found = False
    for i in range(n):
        sum = 0
        for e in range(i,n):
            sum+=arr[e]
            #print(e, 'SUM--> ',sum)
            if sum == total:
                found = True
                return i+1, e+1
    if found == False:
        return -1
    return

if __name__ == '__main__':
    T = int(input())
    for _ in range(T):
        NS = input().split()
        N = int(NS[0])
        S = int(NS[1])
        A = list(map(int, input().split()))
        result = sub_sum(A,S)
        if result != -1:
            i, j = result[0], result[1]
            print(i, j)
        else:
            print(str('-1'))

'''
2
5 12
1 2 3 7 5
10 15
1 2 3 4 5 6 7 8 9 10
'''
